third_party/move/move-model/src/builder/module_builder.rs

```diff
@@ -1319,16 +1321,34 @@ impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
         is_pure
     }
 
-    fn deref_move_fun_types(&mut self, full_name: QualifiedSymbol, spec_fun_idx: usize) {
-        self.parent.spec_fun_table.entry(full_name).and_modify(|e| {
-            assert!(e.len() == 1);
-            e[0].params = e[0]
-                .params
-                .iter()
-                .map(|Parameter(n, ty)| Parameter(*n, ty.skip_reference().clone()))
-                .collect_vec();
-            e[0].result_type = e[0].result_type.skip_reference().clone();
-        });
+    fn deref_move_fun_types(
+        &mut self,
+        full_name: QualifiedSymbol,
+        spec_fun_idx: usize,
+        name_loc: &Loc,
+        name_value: move_symbol_pool::Symbol,
```

Comment:
Maybe just use full_name in the error message? Then you don't need name_value.

Reply:
`full_name` (QualifiedSymbol) does not have a value. An alternative is to pass the function to `deref_move_fun_types` but that creates some issues with borrowing.

Reply:
How about using `full_name.symbol` to retrieve name?

Reply:
full_name.symbol would output the symbol, for example "Symbol(65)" instead of the name of the function. There is no way to extract the function name from the QualifiedSymbol AFAIK.

Reply:
I see, could you try this: `self.symbol_pool().string(full_name.symbol).as_str()` or something similar?

Reply:
I just passed `name: FunctionName` instead and that simplifies the code.